pkg/minikube/tunnel/registry_test.go

```diff
@@ -0,0 +1,275 @@
+	}
+}
+
+func TestPersistentRegistryOperations(t *testing.T) {
+	tcs := []struct {
+		name     string
+		fileName string
+		setup    func(*testing.T, *persistentRegistry)
+		test     func(*testing.T, *persistentRegistry)
+	}{
+		{
+			name:     "Calling List on non-existent registry file should return empty list",
+			fileName: "nonexistent.txt",
+			test: func(t *testing.T, reg *persistentRegistry) {
+				info, e := reg.List()
+				expectedInfo := []*TunnelID{}
+				if !reflect.DeepEqual(info, expectedInfo) || e != nil {
+					t.Errorf("expected %s, nil error, got %s, %s", expectedInfo, info, e)
+				}
+			},
+		},
+		{
+			name:     "Calling Remove on non-existent registry file should return error",
+			fileName: "nonexistent.txt",
+			test: func(t *testing.T, reg *persistentRegistry) {
+				e := reg.Remove(unsafeParseRoute("1.2.3.4", "1.2.3.4/5"))
+				if e == nil {
+					t.Errorf("expected error, got %s", e)
+				}
+			},
+		},
+		{
+			name:     "Calling Register on non-existent registry file should create file",
+			fileName: "nonexistent.txt",
+			test: func(t *testing.T, reg *persistentRegistry) {
+				e := reg.Register(&TunnelID{Route: unsafeParseRoute("1.2.3.4", "1.2.3.4/5")})
+				if e != nil {
+					t.Errorf("expected no error, got %s", e)
+				}
+				fileName := "nonexistent.txt"
+				f, e := os.Open(fileName)
+				if e != nil {
+					t.Errorf("expected file to exist, got: %s", e)
+					return
+				}
+				f.Close()
+				e = os.Remove("nonexistent.txt")
+				if e != nil {
+					t.Errorf("error removing nonexistent.txt: %s", e)
+				}
+			},
+		},
+		{
+			name: "Calling Remove on non-existent tunnel should return error",
+			test: func(t *testing.T, reg *persistentRegistry) {
+				e := reg.Register(&TunnelID{Route: unsafeParseRoute("1.2.3.4", "1.2.3.4/5")})
+				if e != nil {
+					t.Errorf("expected no error, got %s", e)
+				}
+
+				e = reg.Remove(unsafeParseRoute("5.6.7.8", "1.2.3.4/5"))
+				if e == nil {
+					t.Errorf("expected error, got nil")
+				}
+			},
+		},
+		{
+			name: "Register + List should return tunnel info",
+			test: func(t *testing.T, reg *persistentRegistry) {
+				e := reg.Register(&TunnelID{
+					Route:       unsafeParseRoute("1.2.3.4", "1.2.3.4/5"),
+					MachineName: "testmachine",
+					Pid:         1234,
+				})
+				if e != nil {
+					t.Errorf("failed to register: expected no error, got %s", e)
+				}
+
+				tunnelList, err := reg.List()
+				if err != nil {
+					t.Errorf("failed to list: expected no error, got %s", err)
+				}
+
+				expectedList := []*TunnelID{
+					{
+						Route:       unsafeParseRoute("1.2.3.4", "1.2.3.4/5"),
+						MachineName: "testmachine",
+						Pid:         1234,
+					},
+				}
+
+				if len(tunnelList) != 1 || !tunnelList[0].Equal(expectedList[0]) {
+					t.Errorf("\nexpected %+v,\ngot      %+v", expectedList, tunnelList)
+				}
+			},
+		},
+		{
+			name: "Register + Remove + List",
+			test: func(t *testing.T, reg *persistentRegistry) {
```

Comment:
Table driven tests shouldn't need custom per-test logic. Split these out into separate tests.

Reply:
Still this. table driven tests should never contain test-specific logic: http://go/gotestcomments#when-to-use-table-driven-tests